pkg/virt-handler/container-disk/mount.go

```diff
@@ -609,3 +617,70 @@ func (m *mounter) unmountKernelArtifacts(vmi *v1.VirtualMachineInstance) error {
 
 	return fmt.Errorf("kernel artifacts record wasn't found")
 }
+
+func (m *mounter) getContainerDiskPath(vmi *v1.VirtualMachineInstance, volume *v1.Volume, volumeIndex int) (*safepath.Path, error) {
+	sock, err := m.socketPathGetter(vmi, volumeIndex)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := m.podIsolationDetector.DetectForSocket(vmi, sock)
+	if err != nil {
+		return nil, fmt.Errorf("failed to detect socket for containerDisk %v: %v", volume.Name, err)
+	}
+
+	mountPoint, err := isolation.ParentPathForRootMount(m.nodeIsolationResult, res)
+	if err != nil {
+		return nil, fmt.Errorf("failed to detect root mount point of containerDisk %v on the node: %v", volume.Name, err)
+	}
+
+	return containerdisk.GetImage(mountPoint, volume.ContainerDisk.Path)
+}
+
+func computeChecksum(imageFile *safepath.Path) (uint32, error) {
+	var checksum uint32
+
+	err := imageFile.ExecuteNoFollow(func(path string) (err error) {
+		b, err := os.ReadFile(path)
```

Comment:
Is it a good idea to read the whole disk into memory? That could be very expensive, a streaming approach might be the better solution?

```
checksum := crc32.ChecksumIEEE(chunk)
checksum = crc32.Update(checksum, &crc.IEEETable, chunk)
```

Reply:
generally containerdisks are very small (handful of MB), but you do have a point.

I'll change this, thanks for the suggestion.

Reply:
CentOS Stream and RHCOS can reach sizes > 1GiB, so I think it's better to change it. :)

https://quay.io/repository/containerdisks/centos-stream?tab=tags
https://quay.io/repository/containerdisks/rhcos?tab=tags

Reply:
Windows container images are around as well arent they

Reply:
+1

Reply:
fixed, now reads are done in chunks